pallets/msa/src/lib.rs

```diff
@@ -337,6 +400,17 @@ impl<T: Config> Pallet<T> {
 
 		Ok(info)
 	}
+
+	fn ensure_valid_creation_delegate(
+		delegator_key: &T::AccountId,
+		delegate_key: &T::AccountId,
+	) -> Result<Delegate, DispatchError> {
+		let delegate_msa_id = Self::ensure_valid_msa_key(&delegate_key)?.msa_id;
```

Comment:
we cant really use this here because what we want to know is if this key already exist if so than we cannot assign this key. I think we can actually remove this function.

Reply:
there are two things we are checking here

1- if the delegate key has a valid msa account (to ensure it can delegate)
2- if the delegator key has not any registered msa account